links-slackbot/src/main/java/juja/microservices/links/slackbot/util/SlackTextHandler.java

```diff
@@ -2,17 +2,34 @@
 
 import lombok.extern.slf4j.Slf4j;
 
-import java.util.Set;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
- * @author Ivan Shapovalov
+ * @author Ben Novikov
  */
 @Slf4j
 public class SlackTextHandler {
+    /**
+     * URL detection is performed by Slack server, returning links surrounded with angle brackets, then
+     * followed by a pipe character (|) and originally typed version of the URL, if it was not taken as-is.
+     * This method returns bookmarkable links: starting with http, https, ftp and ftps prefixes only.
+     */
+    public List<String> getURLsFromText(String text) {
+        List<String> result = new ArrayList<>();
 
-    public static Set<String> getURLsFromText(String text) {
+        log.debug("Searching for URL links...");
+        Pattern p = Pattern.compile("<(ht|f)tp(s?)://.*?([>|])");
+        Matcher matcher = p.matcher(text);
+        while (matcher.find()) {
+            String found = matcher.group().subSequence(1, matcher.group().length() - 1).toString();
+            log.debug("...found: [{}]", found);
+            result.add(found);
+        }
+        log.debug("URL links found: [{}]", result.size());
```

Comment:
This logs not needed. Spring AOP do this. Replace DEBUG by INFO according to https://github.com/JujaLabs/docs/tree/master/other/workflow